dask_ml/_typing.py

```diff
@@ -0,0 +1,18 @@
+from pathlib import Path
+from typing import TypeVar
+
+import numpy as np
+
+from dask.array import Array
+
+if TYPE_CHECKING:
+    from pandas.core.arrays.base import ExtensionArray  # noqa: F401
```

Comment:
I suspect our algorithms need actual ndarrays or Dask arrays. They have methods like `.reshape` that aren't present on ExtensionArray. Let's be strict for now and add ExtensionArray as appropriate.